php/utils-wp.php

```diff
@@ -319,11 +319,29 @@ function wp_get_table_names( $args, $assoc_args = array() ) {
 	global $wpdb;
 
 	$tables = array();
+
+	// Pre-load tables SQL query with Views restriction if needed.
+	if ( get_flag_value( $assoc_args, 'base-tables-only' ) ) {
+		$tables_sql = $wpdb->prepare( 'SHOW FULL TABLES WHERE Table_Type = %s', 'BASE TABLE' );
```

Comment:
Is this SQL syntax (and the type identifiers) compatible across the entire range SQL servers we're supporting?
If not, we might need to check for the type/version of the SQL server first...

Reply:
To the best of my knowledge it is supported in MySQL 5.6 (docs checked) and MariaDB 10.0 (tested). Does WP CLI support anything else?

Reply:
That should be fine. We'll get issues when something breaks... ;)